core/matrix/bccoo.cpp

```diff
@@ -0,0 +1,789 @@
+
+    // Computation of nnz
+    IndexType nnz = 0;
+    for (const auto& elem : data.nonzeros) {
+        nnz += (elem.value != zero<ValueType>());
+    }
+
+    // Compression. If the initial value is def_value, the default is chosen
+    bccoo::compression compress = this->get_compression();
+    if (compress == bccoo::compression::def_value) {
+        exec->run(bccoo::make_get_default_compression(&compress));
+    }
+
+    // Block partitioning. If the initial value is 0, the default is chosen
+    IndexType block_size = this->get_block_size();
+    if (block_size == 0) {
+        exec->run(bccoo::make_get_default_block_size(&block_size));
+    }
+    IndexType num_blocks = ceildiv(nnz, block_size);
+
+    if (compress == bccoo::compression::element) {
+        // Creation of some components of Bccoo
+        array<IndexType> rows(exec_master, num_blocks);
+        array<size_type> offsets(exec_master, num_blocks + 1);
+
+        // Computation of mem_size (idxs.shf)
+        IndexType* rows_data = rows.get_data();
+        size_type* offsets_data = offsets.get_data();
+        bccoo::compr_idxs<IndexType> idxs;
+        offsets_data[0] = 0;
+        IndexType num = 0;
+        for (const auto& elem : data.nonzeros) {
+            if (elem.value != zero<ValueType>()) {
+                bccoo::put_detect_newblock(rows_data, elem.row - idxs.row,
+                                           idxs);
+                IndexType col_src_res = bccoo::cnt_position_newrow_mat_data(
+                    elem.row, elem.column, idxs);
+                bccoo::cnt_next_position_value(col_src_res, elem.value, idxs);
+                bccoo::put_detect_endblock(offsets_data, block_size, idxs);
+            } else {
+                num++;
+            }
+        }
+
+        // Creation of chunk
+        array<uint8> chunk(exec_master, idxs.shf);
+        uint8* chunk_data = chunk.get_data();
+
+        // Computation of chunk
+        idxs = {};
+        offsets_data[0] = 0;
+        for (const auto& elem : data.nonzeros) {
+            if (elem.value != zero<ValueType>()) {
+                bccoo::put_detect_newblock(rows_data, elem.row - idxs.row,
+                                           idxs);
+                IndexType col_src_res = bccoo::put_position_newrow_mat_data(
+                    elem.row, elem.column, chunk_data, idxs);
+                bccoo::put_next_position_value(chunk_data, col_src_res,
+                                               elem.value, idxs);
+                bccoo::put_detect_endblock(offsets_data, block_size, idxs);
+            }
+        }
+        if (idxs.nblk > 0) {
+            offsets_data[idxs.blk + 1] = idxs.shf;
+        }
+
+        // Creation of the Bccoo object
+        auto tmp =
+            Bccoo::create(exec_master, data.size, std::move(chunk),
+                          std::move(offsets), std::move(rows), nnz, block_size);
+        this->copy_from(std::move(tmp));
```

Comment:
We updated this part in Ginkgo to make it more clear. Please replace all `this->copy_from(std::move(.*))`:
```suggestion
        this->move_from(tmp);
```

Reply:
Done!!